feature/selector.py

```diff
@@ -270,6 +273,102 @@ def _validate(self):
                                                        CatBoostClassifier, CatBoostRegressor)),
                            ValueError("Unknown tree-based estimator" + str(self.estimator)))
 
+    class TextBased(NamedTuple):
+        """
+        # TODO  review the content to double check everything is %100 correct once ALL the implementation is done
+        Text-based method for selecting features/columns from (X_textual, Y).
+        Unlike other selection method, this method operates on textual data, X_textual.
+        In this context, each column/feature in X_textual can be referred to an item with textual description.
+        Similarly, Y represents the labels that are covered by each item.
+        In the test_text.py, X_textual defines as data and Y defines as labels.
+
+        The paper [1] provides a multi-objective optimization problem that is based on
+        set covering formulation. The idea is to select the minimum number of items from X_textual that
+        maximizes the item diversity of the text featurization of items (controlled by the featurization method)
+        while also maximizing the coverage of labels given in Y.
+        If no Y is given, it selects #TODO..
+
+        References:
+        [1] Kadioglu et. al., Optimized Item Selection to Boost Exploration for Recommender Systems, CPAIOR'21
+
+        The parameters corresponding to the methods in the paper [1]
```

Comment:
I think we can remove this block from 294 to 304.. We will set the paper parameters as the default values.